src/truststore/_macos.py

```diff
@@ -280,7 +282,8 @@ def _verify_peercerts_impl(
                 raise MemoryError("Unable to allocate memory!")
 
             # Load all the DER-encoded certificates in the chain
-            for cert_data in cert_chain:
+            ctx_certs: list[bytes] = ssl_context.get_ca_certs(binary_form=True)  # type: ignore[assignment]
```

Comment:
https://docs.python.org/3/library/ssl.html#ssl.SSLContext.get_ca_certs says "The returned list does not contain certificates from capath unless a certificate was requested and loaded by a SSL connection" so I don't think this will include CA certs that were loaded using capath

Also do we need to pass CA certs in a separate array as the 2nd arg for [SecTrustSetAnchorCertificates](https://developer.apple.com/documentation/security/1396098-sectrustsetanchorcertificates) instead of in the same array that contains the chain presented by the server? It seems like that would be needed in order for them to actually be trusted.

Reply:
Regarding `get_ca_certs()` not loading directories eagerly, this is definitely an issue I'll have to open separate to this, my initial looks at Windows APIs didn't show an easy solution so I'll have to look at how this can be done too.

Reply:
This seems to be captured in https://github.com/sethmlarson/truststore/issues/32, we'll likely eagerly load all certificates in a directory.